qa-tests-backend/src/test/groovy/SummaryTest.groovy

```diff
@@ -117,57 +114,49 @@ class SummaryTest extends BaseSpecification {
-                log.info "This diff may help with debug, however deployment names may be different between APIs"
-                List<String> stackroxDeploymentNames = Services.getDeployments(
-                        SearchServiceOuterClass.RawQuery.newBuilder().setQuery(
-                                "Namespace:${ stackroxNamespace.metadata.name }").build()
-                )*.name
-                Javers javers = JaversBuilder.javers()
-                        .withListCompareAlgorithm(ListCompareAlgorithm.AS_SET)
-                        .build()
-                log.info javers.compare(stackroxDeploymentNames, orchestratorNamespace.deploymentCount).prettyPrint()
-                diff = true
+        "Namespace details should match the resources in the orchestrator"
+        for (String ns : orchestratorNamespaces) {
+            withRetry(5, 5) {
+                // Retrieve the namespace details from ACS and the orchestrator
+                def orchestratorNamespaceDetails = orchestrator.getNamespaceDetailsByName(ns)
+                def stackroxNamespaceDetails = NamespaceService.getNamespace(stackroxNamespaces.get(ns))
+
+                if (stackroxNamespaceDetails.numDeployments != orchestratorNamespaceDetails.deploymentCount.size()) {
+                    log.info "There is a difference in the deployment count for namespace ${ns}"
+                    log.info "Stackrox has ${stackroxNamespaceDetails.numDeployments}, " +
+                            "the orchestrator has ${orchestratorNamespaceDetails.deploymentCount.size()}"
+                    log.info "This diff may help with debug, however deployment names may be different between APIs"
+                    List<String> stackroxDeploymentNames = Services.getDeployments(
+                            SearchServiceOuterClass.RawQuery.newBuilder().setQuery(
+                                    "Namespace:${ stackroxNamespaceDetails.metadata.name }").build()
+                    )*.name
+                    Javers javers = JaversBuilder.javers()
+                            .withListCompareAlgorithm(ListCompareAlgorithm.AS_SET)
+                            .build()
+                    log.info javers.compare(stackroxDeploymentNames, orchestratorNamespaceDetails.deploymentCount)
+                            .prettyPrint()
+                }
+                assert stackroxNamespaceDetails.numDeployments == orchestratorNamespaceDetails.deploymentCount.size()
+                assert stackroxNamespaceDetails.metadata.clusterId == ClusterService.getClusterId()
+                assert stackroxNamespaceDetails.metadata.name == orchestratorNamespaceDetails.name
+                assert stackroxNamespaceDetails.metadata.labelsMap == orchestratorNamespaceDetails.labels
+                assert stackroxNamespaceDetails.numSecrets == orchestratorNamespaceDetails.secretsCount
+                assert stackroxNamespaceDetails.numNetworkPolicies == orchestratorNamespaceDetails.networkPolicyCount
```

Comment:
Will we know the namespace if some of those fail? Maybe a log stating the namespace could be added?

Reply:
Makes sense. I've added a log line indicating the namespace we are asserting.